recipes/mongodb_org_repo.rb

```diff
@@ -23,28 +25,37 @@
 # http://www.mongodb.org/downloads#packages
 node.override['mongodb']['package_name'] = 'mongodb-org'
 
+package_version_major = node['mongodb']['package_version'].to_f
+
+package_repo_url = case node['platform']
+                   when 'redhat', 'oracle', 'centos' # ~FC024
+                     "https://repo.mongodb.org/yum/redhat/$releasever/mongodb-org/#{package_version_major}/#{node['kernel']['machine'] =~ /x86_64/ ? 'x86_64' : 'i686'}"
+                   when 'fedora'
+                     "https://repo.mongodb.org/yum/redhat/7/mongodb-org/#{package_version_major}/#{node['kernel']['machine'] =~ /x86_64/ ? 'x86_64' : 'i686'}"
+                   when 'amazon'
+                     "https://repo.mongodb.org/yum/amazon/2013.03/mongodb-org/#{package_version_major}/x86_64/"
+                   end
+
 case node['platform_family']
 when 'debian'
   # Adds the repo: http://www.mongodb.org/display/DOCS/Ubuntu+and+Debian+packages
   apt_repository 'mongodb' do
-    uri "#{node['mongodb']['repo']}/#{node['mongodb']['apt_repo']}"
-    distribution 'dist'
-    components ['10gen']
+    uri node['mongodb']['repo']
+    distribution "#{node['lsb']['codename']}/mongodb-org/#{package_version_major}"
```

Comment:
where did you find this magic combo of facts? it would be useful to have a comment so we can verify it's still true in the future